In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
dataset=pd.read_csv('../CSV Files/googleplaystore.csv')

In [3]:
dataset.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [5]:
dataset.isnull().sum()*100/len(dataset)

App                0.000000
Category           0.000000
Rating            13.596532
Reviews            0.000000
Size               0.000000
Installs           0.000000
Type               0.009224
Price              0.000000
Content Rating     0.009224
Genres             0.000000
Last Updated       0.000000
Current Ver        0.073794
Android Ver        0.027673
dtype: float64

13% of ratings are null
There are some other null values but they are quite negligible

In [6]:
dataset.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

## Category

In [7]:
dataset['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

* One application is of Category '1.9' which is an example of wrong data passed

In [8]:
dataset[dataset['Category']=='1.9']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* Only One App is having such wrong Category

## Ratings

In [9]:
print(sorted(dataset['Rating'].unique()))

[1.0, 1.2, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 4.0, 4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7, nan, 4.8, 4.9, 5.0, 19.0]


* There is some application with 19 as rating. This is an outlier & Any App cannot be rated above 5.0 so we can delete such rows or ask for ratings of concerned apps

In [10]:
dataset[dataset['Rating']>5]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* The same application with Category anomaly is having wrong rating. Better we remove such row

## Reviews

In [11]:
print(sorted(dataset['Reviews'].unique()))

['0', '1', '10', '100', '1000417', '10006', '100082', '100130', '100179', '1002', '1002859', '1002861', '1003269', '10035', '100406', '1004709', '10053186', '10054', '10055521', '1006', '100609', '10067', '1007', '1008', '1008012', '100805', '10088', '10093', '10097', '100994', '100997', '101', '1010', '1011', '10114', '101163', '10117', '1012', '1013', '1013465', '1013635', '1013867', '1013944', '101455', '1014822', '1014846', '1015', '10158', '10159', '1016', '101686', '1017237', '101738', '1017408', '101762', '1018', '101883', '1019', '101957', '10198', '102', '102107', '10212', '10214862', '10216538', '10216997', '10218', '1022', '102215', '102248', '10225', '1024', '102451', '10247', '10249', '1025', '10253', '10256', '102594', '1026', '10270', '102858', '1028637', '1028794', '10291', '102923', '10295', '103', '10306', '103064', '103074', '103078', '1031', '1031045', '10318', '103199', '1032', '1032076', '10323', '1032935', '1033', '103305', '1034', '10341', '10349', '1035', '1035

In [12]:
for i in sorted(dataset['Reviews'].unique()):
    try:
        x=int(i)
    except:
        print(i)
        continue

3.0M


* We need to convert M into 1,000,000 before convert Reviews to Numbers

In [13]:
## Size

In [14]:
dataset.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [15]:
print(dataset['Size'].unique())

['19M' '14M' '8.7M' '25M' '2.8M' '5.6M' '29M' '33M' '3.1M' '28M' '12M'
 '20M' '21M' '37M' '2.7M' '5.5M' '17M' '39M' '31M' '4.2M' '7.0M' '23M'
 '6.0M' '6.1M' '4.6M' '9.2M' '5.2M' '11M' '24M' 'Varies with device'
 '9.4M' '15M' '10M' '1.2M' '26M' '8.0M' '7.9M' '56M' '57M' '35M' '54M'
 '201k' '3.6M' '5.7M' '8.6M' '2.4M' '27M' '2.5M' '16M' '3.4M' '8.9M'
 '3.9M' '2.9M' '38M' '32M' '5.4M' '18M' '1.1M' '2.2M' '4.5M' '9.8M' '52M'
 '9.0M' '6.7M' '30M' '2.6M' '7.1M' '3.7M' '22M' '7.4M' '6.4M' '3.2M'
 '8.2M' '9.9M' '4.9M' '9.5M' '5.0M' '5.9M' '13M' '73M' '6.8M' '3.5M'
 '4.0M' '2.3M' '7.2M' '2.1M' '42M' '7.3M' '9.1M' '55M' '23k' '6.5M' '1.5M'
 '7.5M' '51M' '41M' '48M' '8.5M' '46M' '8.3M' '4.3M' '4.7M' '3.3M' '40M'
 '7.8M' '8.8M' '6.6M' '5.1M' '61M' '66M' '79k' '8.4M' '118k' '44M' '695k'
 '1.6M' '6.2M' '18k' '53M' '1.4M' '3.0M' '5.8M' '3.8M' '9.6M' '45M' '63M'
 '49M' '77M' '4.4M' '4.8M' '70M' '6.9M' '9.3M' '10.0M' '8.1M' '36M' '84M'
 '97M' '2.0M' '1.9M' '1.8M' '5.3M' '47M' '556k' '526k' '76M' '7.6M'

* App Size is in KBs and MBs. We need to use a common metric for App size.
* 'Varies with device' is another anomaly we need to deal with later.

In [16]:
dataset[dataset['Size']=='Varies with device']

App             Category  \
37                                    Floor Plan Creator       ART_AND_DESIGN   
42                            Textgram - write on photos       ART_AND_DESIGN   
52                         Used Cars and Trucks for Sale    AUTO_AND_VEHICLES   
67                                    Ulysse Speedometer    AUTO_AND_VEHICLES   
68                                                REPUVE    AUTO_AND_VEHICLES   
...                                                  ...                  ...   
10713  My Earthquake Alerts - US & Worldwide Earthquakes              WEATHER   
10725                                          Posta App  MAPS_AND_NAVIGATION   
10765                    Chat For Strangers - Video Chat               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   

       Rating Reviews                Size     Installs  Type Price  \
37        4.1   36639  Varies with device   5,000,000+  Free     0   
42        4.4  295221  Varies with device  10,000,000+  Free     0   
52        4.6   17057  Varies with device   1,000,000+  Free     0   
67        4.3   40211  Varies with device   5,000,000+  Free     0   
68        3.9     356  Varies with device     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10713     4.4    3471  Varies with device     100,000+  Free     0   
10725     3.6       8  Varies with device       1,000+  Free     0   
10765     3.4     622  Varies with device     100,000+  Free     0   
10826     4.0   88486  Varies with device   5,000,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   

      Content Rating             Genres        Last Updated  \
37          Everyone       Art & Design       July 14, 2018   
42          Everyone       Art & Design       July 30, 2018   
52          Everyone    Auto & Vehicles       July 30, 2018   
67          Everyone    Auto & Vehicles       July 30, 2018   
68          Everyone    Auto & Vehicles        May 25, 2018   
...              ...                ...                 ...   
10713       Everyone            Weather       July 24, 2018   
10725       Everyone  Maps & Navigation  September 27, 2017   
10765     Mature 17+             Social        May 23, 2018   
10826     Mature 17+             Social      March 23, 2018   
10839     Mature 17+  Books & Reference    January 19, 2015   

              Current Ver         Android Ver  
37     Varies with device        2.3.3 and up  
42     Varies with device  Varies with device  
52     Varies with device  Varies with device  
67     Varies with device  Varies with device  
68     Varies with device  Varies with device  
...                   ...                 ...  
10713  Varies with device  Varies with device  
10725  Varies with device          4.4 and up  
10765  Varies with device  Varies with device  
10826  Varies with device  Varies with device  
10839  Varies with device  Varies with device  

[1695 rows x 13 columns]

* There are 1695 rows with 'Varies with device' which cant be ignored

## Installs

In [17]:
dataset['Installs'].unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0', 'Free'], dtype=object)

* We need to remove '+' from Installation count
* There is an anomaly value ('Free')

In [18]:
dataset[dataset['Installs']=='Free']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* Seriously? Again the same app? 
* Pretty sure the person who formed this data hates 'Life Made WI-Fi Touchscreen Photo Frame' lol.

## Type

In [19]:
dataset['Type'].unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

* What is a '0' type? I bet its again our Super Anomaly App (Life Made WI-Fi Touchscreen Photo Frame)

In [20]:
dataset[dataset['Type']=='0']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* There we go, the prodigal son returns

## Price

In [21]:
dataset['Price'].unique()

array(['0', '$4.99', '$3.99', '$6.99', '$1.49', '$2.99', '$7.99', '$5.99',
       '$3.49', '$1.99', '$9.99', '$7.49', '$0.99', '$9.00', '$5.49',
       '$10.00', '$24.99', '$11.99', '$79.99', '$16.99', '$14.99',
       '$1.00', '$29.99', '$12.99', '$2.49', '$10.99', '$1.50', '$19.99',
       '$15.99', '$33.99', '$74.99', '$39.99', '$3.95', '$4.49', '$1.70',
       '$8.99', '$2.00', '$3.88', '$25.99', '$399.99', '$17.99',
       '$400.00', '$3.02', '$1.76', '$4.84', '$4.77', '$1.61', '$2.50',
       '$1.59', '$6.49', '$1.29', '$5.00', '$13.99', '$299.99', '$379.99',
       '$37.99', '$18.99', '$389.99', '$19.90', '$8.49', '$1.75',
       '$14.00', '$4.85', '$46.99', '$109.99', '$154.99', '$3.08',
       '$2.59', '$4.80', '$1.96', '$19.40', '$3.90', '$4.59', '$15.46',
       '$3.04', '$4.29', '$2.60', '$3.28', '$4.60', '$28.99', '$2.95',
       '$2.90', '$1.97', '$200.00', '$89.99', '$2.56', '$30.99', '$3.61',
       '$394.99', '$1.26', 'Everyone', '$1.20', '$1.04'], dtype=object)

* We need to remove the dollar sign
* Everyone is an outlier

In [22]:
dataset[dataset['Price']=='Everyone']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* Ah Sh!t Here we go again

## Content Rating

In [23]:
dataset['Content Rating'].unique()

array(['Everyone', 'Teen', 'Everyone 10+', 'Mature 17+',
       'Adults only 18+', 'Unrated', nan], dtype=object)

* Nothing weird other than Nan values

## Genres

In [24]:
dataset['Genres'].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

* 'February 11, 2018' outlier

In [25]:
dataset[dataset['Genres']=='February 11, 2018']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

## Last Updated

In [26]:
dataset['Last Updated'].unique()

array(['January 7, 2018', 'January 15, 2018', 'August 1, 2018', ...,
       'January 20, 2014', 'February 16, 2014', 'March 23, 2014'],
      dtype=object)

The usual format consists of 3 parts 'Month' 'Day' 'year'
lets check if there are some exceptions

In [27]:
dataset[~dataset['Last Updated'].apply(lambda x:len(x.split())==3)]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

* Only  Life Made WI-Fi Touchscreen Photo Frame App is having incorrect Last Updated Value

## Current Ver

In [28]:
print(dataset['Current Ver'].unique())

['1.0.0' '2.0.0' '1.2.4' ... '1.0.612928' '0.3.4' '2.0.148.0']


In [29]:
dataset[~dataset['Current Ver'].apply(lambda x:len(str(x).split())==1)]['Current Ver'].unique()

array(['Varies with device', '2.20 Build 02', '3.0.1.11 (Build 311)',
       '4.9.2 (Oasis)', '2.3.34 Prod', 'Cannonau 3.11.11', '3.04 - Beta',
       '4.1.5 (Steglitz)', '1.7.1 (946)', '3.1-2018 – (13)',
       '2.0.0 (35717)', '1.0 Super Ear Hearing', '2.5.0 b665',
       '50.2 lite', '10.9.8 (Play)', '5.2.0 21127', '11.0.269 (m)',
       '3.2.258b ae1da5165', 'Natalia Studio Development',
       '4.1.11 - Hegoa', '5.1.0 free', '5.4 (30)', '0.81 beta',
       'version 0.994', '3.1 Release b2', '1.1 Build 226', '1.2.6 (94)',
       '5.6.2 (US)', '1.3 acremotecontrol', 'Version 1.20',
       '5.29.77 Domain 427', '3.1.0 (build 1210)',
       'Z (Special Edition R9)', 'Cow V3.15', 'Cow V3.11', '1.8 P',
       '7.0 RELEASE', 'BL ONLINE PERSONAL TRAINING 7.9.0',
       'Version 1.3.1', 'Rocksteady 1.3', '2.46 Saber', '7.9.0 (5192)',
       '3.0.0 By Luccas', 'Version 1.6', '5.0.1 build 1',
       'opción de cerrar', '2.2 Free', '2.26 Build 17',
       "1.1 (j'sais pas)", '4.7.0 Phoenix',

* Too much unclean data

## Android Ver

In [30]:
dataset['Android Ver'].unique()

array(['4.0.3 and up', '4.2 and up', '4.4 and up', '2.3 and up',
       '3.0 and up', '4.1 and up', '4.0 and up', '2.3.3 and up',
       'Varies with device', '2.2 and up', '5.0 and up', '6.0 and up',
       '1.6 and up', '1.5 and up', '2.1 and up', '7.0 and up',
       '5.1 and up', '4.3 and up', '4.0.3 - 7.1.1', '2.0 and up',
       '3.2 and up', '4.4W and up', '7.1 and up', '7.0 - 7.1.1',
       '8.0 and up', '5.0 - 8.0', '3.1 and up', '2.0.1 and up',
       '4.1 - 7.1.1', nan, '5.0 - 6.0', '1.0 and up', '2.2 - 7.1.1',
       '5.0 - 7.1.1'], dtype=object)

* We can convert this into numeric value by splitting string and taking the minimum android version value

* Summarizing everything we learnt so far

* PS= Handle the prodigy son row
|Feature   	|Task    	|
|---	|---	|
|App   	|  NA 	|
|Category   	|PS   	|
|Rating   	|PS & NaN   	|
|Reviews   	| Convert M to numerical value   	|
|Size   	| COnvert KB to MB & handle varies with device   	|
|Installs   	| Remove '+', PS  	|
|Type   	| PS   	|
|Price   	| Remove $ & PS   	|
|Content Rating   	| NAN   	|
|Genres   	|PS   	|
|Last Updated   	|PS   	|
|Current Version   	| Too much unclean data, gotta figure out atleast a floating number denoting version   	|
|Android Version   	|String to Numeric   	|


* Now there is another csv file containing user reviews & sentiments for different applications 

In [40]:
dataset_reviews=pd.read_csv('../CSV Files/googleplaystore_user_reviews.csv')

In [57]:
no_reviews=0
dataset_review_apps=dataset_reviews['App'].unique()
for i in dataset['App']:
    if i in dataset_review_apps:
        continue
    no_reviews+=1
print(no_reviews,"Apps have no reviews out of Total",len(dataset),'Apps')
    

9309 Apps have no reviews out of Total 10841 Apps


* As The Reviews csv file does not contain for every App in our dataset, we will ignore this csv for now